packages/sanity/src/core/releases/tool/detail/useBundleDocuments.ts

```diff
@@ -144,6 +148,36 @@ const getActiveReleaseDocumentsObservable = ({
             return documentPreviewStore
               .observeForPreview(document, schemaType, {perspective: [releaseId]})
               .pipe(
+                switchMap((value) => {
+                  if (value.snapshot) {
+                    return of(value)
+                  }
+
+                  // remove the first stack (since we want to get the previous perspective)
```

Comment:
Maybe add a bit of contextual info to make it clearer why we're doing this?
E.g.
> If we don't receive a snapshot here, it means the document will be unpublished in the release. If that's the case, we want to preview it as it's latest known version instead (e.g. one stack level up).